storage/range.go

```diff
@@ -819,21 +864,9 @@ func (r *Range) applyRaftCommand(ctx context.Context, index uint64, originNodeID
 			if log.V(1) {
 				log.Infoc(ctx, "found response cache entry for %+v", args.Header().CmdID)
 			}
-			return reply.Header().GoError()
-		} else if ok && err != nil {
-			newErr := util.Errorf("unable to read result for %+v from the response cache: %s", args, err)
-			reply.Header().SetGoError(newErr)
-			return newErr
-		}
-	}
-
-	// Verify the leader lease is held; Note that we don't require the
-	// leader lease when trying to grant the leader lease!
-	if _, ok := args.(*proto.InternalLeaderLeaseRequest); !ok {
-		if !r.verifyLeaderLease(originNodeID, header.Timestamp) {
-			err := r.newNotLeaderError()
-			reply.Header().SetGoError(err)
 			return err
+		} else if ok && err != nil {
+			return newReplicaCorruptionError("could not read from response cache", err)
```

Comment:
yes, that shouldn't happen. I'll make sure the first replicaCorruptionError seals the deal.